directory/tests/test_directory.py

```diff
@@ -0,0 +1,246 @@
+        else:
+            # 12 files in 'temp_dir'
+            aggregator.assert_metric(
+                "system.disk.directory.files",
+                tags=dir_tags, count=1, value=12)
+
+    # Raises when coverage < 100%
+    aggregator.metrics_asserted_pct == 100.0
+
+
+@pytest.mark.integration
+def test_file_metrics(aggregator):
+    """
+    File metric coverage
+    """
+    config_stubs = get_config_stubs(temp_dir, filegauges=True)
+
+    config = {
+        'instances': config_stubs
+    }
+
+    for config in config_stubs:
+        aggregator.reset()
+        dir_check.check(config)
+        dirtagname = config.get('dirtagname', "name")
+        name = config.get('name', temp_dir)
+        filetagname = config.get('filetagname', "filename")
+        dir_tags = [dirtagname + ":%s" % name, 'optional:tag1']
+
+        # File metrics
+        for mname in FILE_METRICS:
+            if config.get('pattern') != "file_*":
+                # 2 '*.log' files in 'temp_dir'
+                for i in xrange(1, 3):
+                    file_tag = [
+                        filetagname + ":%s" % os.path.normpath(
+                            temp_dir + "/log_" + str(i) + ".log")
+                        ]
+                    aggregator.assert_metric(
+                        mname,
+                        tags=dir_tags + file_tag,
+                        count=1)
+
+            if config.get('pattern') != "*.log":
+                # Files in 'temp_dir'
+                for i in xrange(0, 10):
+                    file_tag = [
+                        filetagname + ":%s" % os.path.normpath(
+                            temp_dir + "/file_" + str(i))
+                        ]
+                    aggregator.assert_metric(
+                        mname,
+                        tags=dir_tags + file_tag,
+                        count=1)
+
+            if not config.get('pattern'):
+                # Files in 'temp_dir/subfolder'
+                if config.get('recursive'):
+                    for i in xrange(0, 5):
+                        file_tag = [
+                            filetagname + ":%s" % os.path.normpath(
+                                temp_dir + "/subfolder" + "/file_" + str(i))
+                            ]
+                        aggregator.assert_metric(
+                            mname,
+                            tags=dir_tags + file_tag,
+                            count=1)
+
+        # Common metrics
+        for mname in COMMON_METRICS:
+            aggregator.assert_metric(mname, tags=dir_tags, count=1)
+
+        # Raises when coverage < 100%
+        assert aggregator.metrics_asserted_pct == 100.0
+
+
+@pytest.mark.integration
+def test_non_existent_directory(aggregator):
+    """
+    Missing or inaccessible directory coverage.
+    """
+    config = {'instances': [{'directory': '/non-existent/directory'}]}
+    with pytest.raises(Exception):
+        dir_check.check(config)
+
+
+@pytest.mark.integration
```

Comment:
get rid of these. these `@pytest.mark.integration` mean it's not running. Don't mark them with anything

Reply:
After talking further, since tox is running `pytest .` it should be running all tests despite the metadata marker applied to them. Though since they are all integration tests, I removed them.